core/services/mongodb-remote/src/test/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/SyncUnitTestHarness.kt

```diff
@@ -354,8 +366,16 @@ class SyncUnitTestHarness : Closeable {
                 bsonDocumentCodec)
         }
 
-        override fun waitForEvent() {
-            assertTrue(eventSemaphore?.tryAcquire(10, TimeUnit.SECONDS) ?: true)
+        override fun waitForEvents(amount: Int) {
+            waitLock.lock()
+            changeEventListener.totalEventsToAccumulate = amount
```

Comment:
[q] Why is this here?

Reply:
This is to give the harness the ability to wait for multiple events. The accumulator actually does nothing– could just be a counter– as the actual events are later accumulated via capturing the values from the spy.

This is being used to test bulk writes, as multiple change events are emitted from a single method call.